LocalNotifications/Sources/LocalNotifications/LocalNotifications.swift

```diff
@@ -0,0 +1,136 @@
+                return
+            }
+
+            do {
+                GD.print("Requesting Authorisation")
+                let authorizationGranted = try await notificationCenter.requestAuthorization(options: authorizationOptions)
+                self.permitted = authorizationGranted
+                onComplete.callDeferred()
+            } catch {
+                GD.print("Authorisation request failed. \(error.localizedDescription)")
+                self.permitted = false
+            }
+        }
+    }
+    
+    @Callable
+    func isPermitted() -> Bool {
+        self.permitted
+    }
+
+    @Callable
+    func wasNotificationAccepted() -> Bool {
+        self.notificationAccepted
+    }
+
+    @Callable
+    func removeAllNotifications() {
+        let notificationCenter = UNUserNotificationCenter.current()
+        let pending = notificationCenter.removeAllPendingNotificationRequests()
+    }
+
+    @Callable
+    func removeNotification(uid:String) {
+        let notificationCenter = UNUserNotificationCenter.current()
+        notificationCenter.removePendingNotificationRequests(withIdentifiers: [uid])
+    }
+
+    /**
+        Request that a notification banner be displayed in the future, as long
+        as the user is not inside the application. `reqeustPermission()` must
+        be called first.
+
+        - parameter uid: Random string or identifier used to delete this notification later if needed
+        - parameter title: title of the local notification.
+        - parameter body: text body of the local notification.
+        - parameter seconds: display notification in how many seconds time.
+        - returns no value is instantly expected. onComplete read the variable `notificationAccepted`
+     */
+    @Callable
+    func schedule(_ uid:String, _ title:String, _ body:String, _ seconds:Int, _ onComplete:Callable) {
+        Task {
+            if await hasPendingNotification(uid:uid) {
+                GD.print("Notification already exists with UID \(uid). Rescheduling.")
+                await removeNotification(uid:uid)
+            }
+            self.notificationAccepted = false
+
+            let content = UNMutableNotificationContent()
+            content.title = title
+            content.body = body
+            content.sound = UNNotificationSound.default
+        
+            let trigger = UNTimeIntervalNotificationTrigger(timeInterval:  TimeInterval(seconds), repeats: false)
+            let request = UNNotificationRequest(identifier: uid, content: content, trigger: trigger)
+
+            do {
+                try await UNUserNotificationCenter.current().add(request)
+                GD.print("Scheduled notification \(uid).")
+                notificationAccepted = true
+                onComplete.callDeferred()
+            } catch {
+                GD.print("Error scheduling: \(error.localizedDescription)")
+                notificationAccepted = false
+                onComplete.callDeferred()
```

Comment:
Using callDeferred is nice and simplifies things on the Godot side, but it might be better to return an error value if something fails so the user can react to the problem. The standard Godot way is to return 0 as an ok and >0 for errors.

You should be able to use onComplete.bind() / bindv() to add parameters even though callDeferred doesn't take any.

I fell back to using callv() because I didn't get the .bind() functions to work, it might be a bug in SwiftGodot or I did something wrong, will investigate on my end as using callDeferred is probably preferred